### PR TITLE
Refine raw model for template part-1

### PR DESCRIPTION
**Pull request by OsmondJiang**
1. reuse page model to generate raw model as many as possible
2. add test to check generated raw model, in case we will break in the future changing, will remove it later once our page model become raw model
3. fix some bugs

### CODE REVIEW

File: docs/specs/legacy.yml [hunk 1]
```diff
@@ -71,6 +71,67 @@ outputs:
...
+        "api_name": ["API Name"],
+        "ms.author": "yufeih",
+        "depot_name": ".",
+        "site_name": "Docs",
+        "__global": {},
+        "conceptual": "<p><a href=\"d.png\">d</a></p>\n",
+        "_path": "c.json",
+        "toc_rel": "toc.json",
+        "word_count": 2,
+        "wordCount": 2,
```
Comment (OsmondJiang) [inline docs/specs/legacy.yml]: two word_count and two locale here

File: docs/specs/legacy.yml [hunk 2]
```diff
@@ -71,6 +71,67 @@ outputs:
...
+        "title": "Title",
+        "rawTitle": "<h1 id=\"title\">Title</h1>",
+        "_op_canonicalUrlPrefix": "https://docs.com/en-us/./",
```
Comment (OsmondJiang) [inline docs/specs/legacy.yml]: this is somehow strange

File: src/docfx/template/TemplateEngine.cs [hunk 3]
```diff
@@ -301,7 +295,7 @@ private static JObject CreateMetadata(JObject rawMetadata)
 
         private static string CreateHtmlMetaTags(JObject metadata)
         {
-            var result = new StringBuilder();
+            var meta = new List<string>();
 
             foreach (var (key, value) in metadata)
```
Comment (yufeih) [inline src/docfx/template/TemplateEngine.cs]: order here

File: src/docfx/docset/Document.cs [hunk 4]
```diff
@@ -471,7 +471,8 @@ private static bool IsValidRelativePath(string path)
             // get set path from site path
             // site path doesn't contain version info according to the output spec
             var sitePathWithoutExtension = Path.Combine(Path.GetDirectoryName(SitePath), Path.GetFileNameWithoutExtension(SitePath));
-            var sitePath = PathUtility.NormalizeFile(Path.GetRelativePath(Docset.Config.DocumentId.SiteBasePath, sitePathWithoutExtension));
+            var siteBasePath = Docset.Config.Output.LowerCaseUrl ? Docset.Config.DocumentId.SiteBasePath.ToLowerInvariant() : Docset.Config.DocumentId.SiteBasePath;
```
Comment (yufeih) [inline src/docfx/docset/Document.cs]: Is there a bug for this one? suggest not change this in this PR.

Comment (OsmondJiang) [reply src/docfx/docset/Document.cs]: this bug is found in my test, so i need fix it to make my test pass in linux.

**Pull request merged**